Python/frame.c

```diff
@@ -115,6 +115,17 @@ take_ownership(PyFrameObject *f, _PyInterpreterFrame *frame)
     }
 }
 
+void
+_PyFrame_ClearLocals(_PyInterpreterFrame *frame)
+{
+    assert(frame->stacktop >= 0);
+    for (int i = 0; i < frame->stacktop; i++) {
```

Comment:
`frame->stacktop = 0` needs to happen before decref'ing the locals, otherwise the GC might see invalid objects